### PR TITLE
SALTO-5010: [Zendesk] Fix race condition when creating new variants for a dynamic_content_item

### PR DESCRIPTION
**Pull request by DanielBarros1**
When creating new variants, we send 2 requests:
1. to modify the parent item
2. to create the variants.
This can create a race condition in Zendesk that returns an error.
By ordering the deployments correctly + ignoring the variants field on the item, we make sure the 2 requests don't contradict each other

---

_Test Plan_:

Cases i tested manually:
1. Add 2 variants and modify item
2. Create item + variant
3. Delete variant
4. Delete item + variant
5. Modify variant
6. Modify item

---

_Release Notes_: 
Zendesk: 
Fix race condition when creating new variants for a dynamic_content_item

---

_User Notifications_: 
None

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/dynamic_content.ts [hunk 1]
```diff
@@ -71,12 +71,25 @@ const filterCreator: FilterCreator = ({ config, client }) => ({
...
+      )
+      const [itemRemovalChanges, itemNonRemovalChanges] = _.partition(itemChanges, isRemovalChange)
+
       const deployResult = await deployChangesByGroups(
-        [nonRemovalChanges, removalChanges] as Change<InstanceElement>[][],
+        [
+          variantAdditionChanges,
+          itemNonRemovalChanges,
+          variantModificationChanges,
+          itemRemovalChanges,
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/dynamic_content.ts:90]: you don't get an error if you remove an item and its variant after?

File: packages/zendesk-adapter/src/filters/dynamic_content.ts [hunk 2]
```diff
@@ -71,12 +71,25 @@ const filterCreator: FilterCreator = ({ config, client }) => ({
       relevantChanges,
       change => getChangeData(change).elemID.typeName === DYNAMIC_CONTENT_ITEM_TYPE_NAME,
     )
+
     if (itemChanges.length === 0 || itemChanges.every(isModificationChange)) {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/dynamic_content.ts:75]: we are keeping the every? what happens if we have and addition of an item and a modification in the same deploy? do they get to this filter in different groups?

**Review by edenhassid — COMMENTED** [review]
looks good!! 💯 🤩 added some small questions

**Review by edenhassid — COMMENTED** [review]
did you also test modifying an item by adding a variant and removing another variant?

File: packages/zendesk-adapter/src/filters/dynamic_content.ts (hunk 2 above)
Comment (DanielBarros1) [reply packages/zendesk-adapter/src/filters/dynamic_content.ts:75]: I checked a deployment with both addition and modification and it seems like it's working like you suggested - two different groups, so it works well i guess :)

File: packages/zendesk-adapter/src/filters/dynamic_content.ts (hunk 1 above)
Comment (DanielBarros1) [reply packages/zendesk-adapter/src/filters/dynamic_content.ts:90]: Nope! deleted a variant and an item and the deployment didn't throw

File: packages/zendesk-adapter/src/filters/dynamic_content.ts (hunk 2 above)
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/dynamic_content.ts:75]: i think that if we get here then there is only one item in the change group, and then we don't need the `itemRemovalChanges,` in the code. WDYT?

Comment (DanielBarros1) [reply packages/zendesk-adapter/src/filters/dynamic_content.ts:75]: I think you're probably right, but i'd probably opt to leave it as is - this way the ordering of the deployment is as explicit and verbose as possible, saves us from having to add a bunch of comments explaining the precise circumstances.

**Review by edenhassid — APPROVED** [review]
LGTM!! 😄 💯

**Pull request merged**